### DISCUSSION TITLE
Where to find an offline-mode tutorial?

### DISCUSSION THREAD

**Discussion by realtebo** [quasarframework/quasar#14296] [CLI - PWA mode]
My app has received the request from the customer to be able to "work" offline. 

It's about 99% a catalogue, so we could pre-cache 100% of assets. 

There is a cart section where were can save to localstorage if not online.

But it's the 'network first' mode for the rest of app content my problem... 
I'm not able to find a simple PWA offline tutorial *specific for Quasar*

**Comment by TobyMosque** [comment]
that look like a offline first app, so u source of truth would be available offline, your local database would sync with your API when there is a internet connection avaliable.

personally, i would recommend the RxDB for that task.
https://rxdb.info/offline-first.html

**Reply by realtebo** [reply]
Will that library work both on webapp AND android/ios apps?

**Reply by realtebo** [reply]
Also, ok, this can solve our problem of 'where' save data

But how to make the app, routing, etc works EVEN if online, it's a quasar+offline problem  I think I should resolve using PWA feature "offline", but I am unable to find a good tutorial on how to let a webapp works offline !

**Reply by TobyMosque** [reply]
Only the API calls are network-only, all asserts are network-first/cache-first, since u app hadn't API calls, so that will work offline.

But how do I can build a app without API calls? just use a offline-first database, as RxDB, firebase (with persistence enabled), etc... All your call will be resolved locally... That libs (RxDB, firebase, etc) will keep the local database on sync with your API, but we don't need to access the API/backend directly.
